os/tools/memstats_xtensa.py

```diff
@@ -0,0 +1,361 @@
+#!/usr/bin/env python
+###########################################################################
+#
+# Copyright 2019 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+import sys
+import re
+from sets import Set
+from optparse import OptionParser
+
+parser = OptionParser()
+parser.add_option("-f", "--file", dest="infilename", help="map FILE needed to be parsed", metavar="INPUT_FILE")
+parser.add_option("-o", "--output",dest="output", help="Output written to this file. Default is stdout.", metavar = "OUTPUT_FILE")
+parser.add_option("-a", "--all", action="store_true", dest="all", help="Print all information.", default = False)
+parser.add_option("-s", "--size", action="store_true", dest="totsize", help="Print total size of data segments.", default=False)
+parser.add_option("-l", "--libsize", action="store_true", dest="libsize", help="Print Library Sizes.", default = False)
+parser.add_option("-d", "--details", action="store_true", dest="details", help="Print details of Object Sizes.",default = False)
+#Example: memstats_xtensa.py <option {-a / -s / -l / -d}> -f tinyara.map -o memory.txt.
+
+(options, args) = parser.parse_args()
+if not options.infilename:
+        parser.print_help()
+        sys.exit(1)
+
+if (not options.all) and (not options.totsize) and (not options.libsize) and (not options.details):
+        parser.print_help()
+        sys.exit(1)
+
+infile = open(options.infilename,'r')
+if options.output:
+        sys.stdout = open(options.output,'w')
+first = 0
+currentSymbol = ""
+level1 = {}
+action = 0
+subSymbol = 0
+libObject = ""
+librarySize = {}
+objectSize = {}
+allLibraries = {}
+
+Required = {
+".text":[],
+".data":[],
+".bss":[],
+}
+
+Symnamemap = {
+".iram0.vectors":[],
+".iram0.text":[],
+".flash.text":[],
+".dram0.data":[],
+".dram0.bss":[],
+".flash.rodata":[]
+}
+
+key_list = Required.keys()
+Symnamemap[".iram0.vectors"] = key_list[2]; #text seg 
+Symnamemap["flash.rodata"] = key_list[2];   #text seg
```

Comment:
flash.rodata seems better included in data seg, what's your opinion?

Reply:
rodata and text section both just need load once,
so mapped them to one segment.

reference https://lwn.net/Articles/531148/

Reply:
looks good to me